src/crypto/sha1.cpp

```diff
@@ -343,13 +352,44 @@ void Transform(uint32_t* s, const unsigned char* chunk)
     s[2] += c;
     s[3] += d;
     s[4] += e;
+}
 
+/** Define SHA1 hardware */
+#if defined(__linux__)
+#define HWCAP_SHA1  (1<<5)
+#include <sys/auxv.h>
+#elif defined(__WIN64__)
+#include <intrin.h>
+bool isAVX (void) {
+  int cpuinfo[4];
+  __cpuid(cpuinfo, 1);
+  return ((cpuinfo[2] & (1 << 28)) != 0);
+}
 #endif
 
+/** Define a function pointer for Transform */
+void (*transform_ptr) (uint32_t*, const unsigned char*) = &Transform;
+
+/** Initialize the function pointer */
+void inline Initialize_transform_ptr(void)
+{
+// Override the function pointer for ARMV8/AVX2
+#if (defined(USE_ARMV8) || defined(USE_ARMV82))
+    if (getauxval(AT_HWCAP) && HWCAP_SHA1)
```

Comment:
I think this should be `&` instead of `&&` here, below and in the other `sha*.cpp` files?

i.e.:
```cpp
if (getauxval(AT_HWCAP) & HWCAP_SHA1) 
```